Integrations/FeedO365/FeedO365.py

```diff
@@ -0,0 +1,204 @@
+            response = requests.get(
+                url=url,
+                verify=self._verify
+            )
+            try:
+                response.raise_for_status()
+                data = response.json()
+                result.extend([i for i in data if 'ips' in i or 'urls' in i])  # filter empty entries
+            except ValueError as err:
+                raise ValueError(f'Could not parse returned data to Json. \n\nError massage: {err}')
+        return result
+
+
+def batch_indicators(sequence, batch_size=1) -> List:
+    """Batch the indicators to balance load on the server.
+
+    Args:
+        sequence: all items
+        batch_size: how many items to batch
+
+    Returns:
+        A List of batch_size of items.
+    """
+    sequence_length = len(sequence)
+    for i in range(0, sequence_length, batch_size):
+        yield sequence[i:min(i + batch_size, sequence_length)]
+
+
+def test_module(client: Client) -> Tuple[str, Dict, Dict]:
+    """Builds the iterator to check that the feed is accessible.
+    Args:
+        client: Client object.
+
+    Returns:
+        Outputs.
+    """
+    client.build_iterator()
+    return 'ok', {}, {}
+
+
+def get_indicators_command(client: Client, indicator_type: str) -> Tuple[str, Dict, Dict]:
+    """Retrieves indicators from the feed to the war-room.
+
+    Args:
+        client: Client object with request
+        indicator_type: indicator_type to be retrieved.
+
+    Returns:
+        Outputs.
+    """
+    iterator = client.build_iterator()
+    indicator_type_lower = indicator_type.lower()
+    indicators = []
+    raw_response = []
+    limit = int(demisto.args().get('limit')) if 'limit' in demisto.args() else 100
+    iterator = [i for i in iterator if indicator_type_lower in i]  # filter indicator_type specific entries
+    iterator = iterator[:limit]
+    for item in iterator:
+        values = item.get(indicator_type_lower)
+        raw_json = {'type': indicator_type[:-1]}
+        if values:
+            for value in values:
+                raw_json['value'] = value
+                indicators.append({
+                    "Value": value,
+                    "Type": indicator_type[:-1],
+                    'rawJSON': {"Value": value, "Type": indicator_type[:-1]}
+                })
+                raw_response.append(raw_json)
+    human_readable = tableToMarkdown('Indicators from O365 Feed:', indicators,
+                                     headers=['Value', 'Type'], removeNull=True)
+
+    return human_readable, {'O365.Indicator': indicators}, raw_response
+
+
+def fetch_indicators_command(client: Client) -> List[Dict]:
+    """Fetches indicators from the feed to the indicators tab.
+
+    Args:
+        client: Client object with request
+
+    Returns:
+        Indicators.
+    """
+    indicator_type = client.indicator_type
+    indicator_type_lower = indicator_type.lower()
+    iterator = client.build_iterator()
+    indicators = []
+    for item in iterator:
```

Comment:
can't fetch indicators and get indicators use the same sub function ti get the indicators and handle the raw response?

Reply:
If i want to limit the results prior to iterating all of them, i think it's less confusing to split it.